test/module/irohad/consensus/yac/peer_orderer_test.cpp

```diff
@@ -49,10 +53,13 @@ class YacPeerOrdererTest : public ::testing::Test {
     orderer = PeerOrdererImpl(wsv);
   }
 
-  std::vector<iroha::model::Peer> peers = [] {
-    std::vector<iroha::model::Peer> result;
+  std::vector<std::shared_ptr<shared_model::interface::Peer>> peers = [] {
+    std::vector<std::shared_ptr<shared_model::interface::Peer>> result;
     for (size_t i = 1; i <= N_PEERS; ++i) {
-      result.push_back(iroha::consensus::yac::mk_peer(std::to_string(i)));
+      auto peer = std::shared_ptr<shared_model::interface::Peer>(shared_model::proto::PeerBuilder()
+                      .address(std::to_string(i)).pubkey(shared_model::interface::types::PubkeyType(std::string(32, '0')))
+                      .build().copy());
```

Comment:
Why is copy here?

Reply:
To allocate a new object on the heap. Proto builders return objects on the stack